src/driver.ts

```diff
@@ -262,20 +229,14 @@ export abstract class CMakeDriver implements vscode.Disposable {
     if (this._kit.environmentVariables) {
       util.objectPairs(this._kit.environmentVariables).forEach(([k, v]) => this._kitEnvironmentVariables.set(k, v));
     }
-    switch (this._kit.type) {
-    case 'vsKit': {
+    if (this._kit.visualStudio && this._kit.visualStudioArchitecture) {
       const vars = await getVSKitEnvironment(this._kit);
```

Comment:
Could we move this visual studio dependend environmentVariables to code which loads kits.
The driver makes no difference between VSKitEnvironment variables and kitEnvironementVariables.

The created error is a more important information on loading of kits.